src/Implement/UnitTestHttpListener.cpp

```diff
@@ -0,0 +1,47 @@
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "HttpListener.h"
+
+// Testing HttpListener mock class
+class HttpListenerMock : public HttpListener {
+ private:
+  static HttpListenerMock* test_listener;
+  HttpListenerMock(){
+
+  };
+
+ public:
+  static HttpListenerMock* GetInstance() {
+    HttpListener::GetInstance();
+    if (test_listener == nullptr) {
+      test_listener = new HttpListenerMock();
+    }
+    return test_listener;
+  }
+  ::testing::MockFunction<void(const int&, const std::string&)> mock_callback;
+  // MOCK_METHOD(void, callback, (const int &sock_fd, const std::string &data),
+  // (override));
+  MOCK_METHOD(void, ThreadListen, (), (const));
+  MOCK_METHOD(void, BindPort, (const int& port_num), (const));
+};
+
+// Init static variable for mock
+HttpListenerMock* HttpListenerMock::test_listener = nullptr;
+
+// Test waiting for request and do the callback
+TEST(HttpListener_MOCK, Listening) {
```

Comment:
As discussed in class, we want to test HttpListener's functionality, not the mock.

Reply:
This is still not testing the HttpListener's functionality.

You need to use a TCP Client to connect to port 9000, then check if test_callback is called.

Reply:
Please implement the unit test as discussed in class.

Reply:
Done